# HG changeset patch
# User ysuenaga
# Date 1432906184 -32400
#      Fri May 29 22:29:44 2015 +0900
# Node ID 42f30231d979516785eb6cca84399be2797449b3
# Parent  cf23177e21ca6f49bdfcecaf2020db81d8d659a1
8081475, PR2495: SystemTap does not work when JDK is compiled with GCC 5
Summary: libjvm.so which is generated by GCC 5 does not have .note.stapsdt section as dtrace was disabled due to incorrect version check
Reviewed-by: dholmes, coleenp

diff -r cf23177e21ca -r 42f30231d979 make/linux/makefiles/dtrace.make
--- openjdk/hotspot/make/linux/makefiles/dtrace.make	Wed Jul 01 18:36:36 2015 +0100
+++ openjdk/hotspot/make/linux/makefiles/dtrace.make	Fri May 29 22:29:44 2015 +0900
@@ -31,8 +31,8 @@
 REASON = "This JDK does not support SDT probes"
 else
 
-# We need a recent GCC for the default
-ifeq "$(shell expr \( $(CC_VER_MAJOR) \>= 4 \) \& \( $(CC_VER_MINOR) \>= 4 \) )" "0"
+# We need a recent GCC for the default (4.4 or later)
+ifeq "$(shell expr \( \( $(CC_VER_MAJOR) = 4 \) \& \( $(CC_VER_MINOR) \>= 4 \) \) \| \( $(CC_VER_MAJOR) \>= 5 \) )" "0"
 REASON = "gcc version is too old"
 else
 
